pkg/cmd/controller/controller.go

```diff
@@ -58,3 +63,17 @@ func NewCmdController(commonOpts *opts.CommonOptions) *cobra.Command {
 func (o *ControllerOptions) Run() error {
 	return o.Cmd.Help()
 }
+
+func (o *ControllerOptions) shouldRegisterCRD(jxClient versioned.Interface, devNs string) (bool, error) {
+	devEnv, err := jxClient.JenkinsV1().Environments(devNs).Get("dev", metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	requirements, err := config2.GetRequirementsConfigFromTeamSettings(&devEnv.Spec.TeamSettings)
```

Comment:
What's about we add an explicit `IsBootCluster` to `config2` or another appropriate place? It took me a while to figure out the intention here since there is the implicit assumption that if there is no requirements file in the team settings, we are dealing with a non-boot cluster.

Reply:
The problem with that function is that it wouldn't be deterministic depending on the scope where it's called.

For example if it's called at the beginning of boot, it will return false, but in the scope of a controller, it will always be deterministic.

Reply:
You could make it a function of this controller then? If it were not for the title of the pull request I would not know what this could is about. More so since as you just pointed out there are further constraints. 

Is this also something we need only for clusters created by `jx create cluster`? If so, is it worth adding a comment that this is maybe temporary or why we don't want to do this in a boot cluster?

Reply:
Whether it was created with `jx create cluster` is irrelevant. When we boot a cluster, que add the requirements to the TeamSettings at some point in the process, this is what I'm looking for.

This while change comes because registering CRDs on a controller in a boot cluster doesn't make sense but I'm still not sure about a `jx install` cluster.

Maybe we could remove it altogether?

Reply:
> Maybe we could remove it altogether?

That would be the preferred option, imo. Have you tried?